terraform/032-db-backup/main.tf

```diff
@@ -0,0 +1,102 @@
+/*
+ * Create S3 bucket for storing backups
+ */
+resource "aws_s3_bucket" "backup" {
+  bucket        = "${var.idp_name}-${var.app_name}-${var.app_env}"
+  acl           = "private"
+  force_destroy = true
+
+  versioning {
+    enabled = true
+  }
+
+  lifecycle_rule {
+    id      = "delete-old-versions"
+    enabled = true
+
+    noncurrent_version_expiration {
+      days = 30
+    }
+  }
+
+  tags {
+    idp_name = "${var.idp_name}"
+    app_name = "${var.app_name}"
+    app_env  = "${var.app_env}"
+  }
+}
+
+/*
+ * Create user for putting backup files into the bucket
+ */
+resource "aws_iam_user" "backup" {
+  name = "db-backup-${var.idp_name}-${var.app_env}"
+}
+
+resource "aws_iam_access_key" "backup" {
+  user = "${aws_iam_user.backup.name}"
+}
+
+resource "aws_iam_user_policy" "backup" {
+  name = "S3-DB-Backup"
+  user = "${aws_iam_user.backup.name}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject"
```

Comment:
Will this need a different/expanded policy when in `restore` mode?

Reply:
it would be, or a different user entirely